src/allocation_engine/interval_tree.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright (C) 2022 Alibaba Cloud. All rights reserved.
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2
+
+/// Policy for resource allocation.
+#[derive(Copy, Clone, Debug)]
```

Comment:
We should add `PartialEq` as well. You can check the guidelines for APIs here: https://rust-lang.github.io/api-guidelines/interoperability.html#types-eagerly-implement-common-traits-c-common-traits. We can leave out Display for now, and Hash, but the other ones are trivial to add as we can just use derive.

Reply:
Same for Constraint.